pe_exceptions/core/pe_base_exception.hpp

```diff
@@ -0,0 +1,30 @@
+#ifndef PE_BASE_EXCEPTION_HPP
+#define PE_BASE_EXCEPTION_HPP
+
+#include <fstream>
+#include <stdexcept>
+#include <string_view>
+
+namespace peach::detail
+{
+  class pe_base_exception : virtual public std::runtime_error
+  {
+  public:
+    pe_base_exception( ) = default;
+
+    // clang-format off
+    pe_base_exception( const pe_base_exception& ) noexcept = default;
+    pe_base_exception( pe_base_exception&& ) noexcept = delete;
+
+    pe_base_exception& operator=( pe_base_exception&& ) = delete;
+    pe_base_exception& operator=( const pe_base_exception& ) = delete;
+    // clang-format on
+
+    virtual bool output_to_file( std::ofstream& output_file ) const = 0;
+    virtual void print_to_console( ) const = 0;
```

Comment:
You should just have an std::ostream& operator<<(...)

Reply:
fixed in 7fac1d4, thanks a ton!